source/features/more-dropdown.tsx

```diff
@@ -31,10 +31,32 @@ async function init(): Promise<void> {
 		createDropdown();
 	}
 
+	let compareUrl = `/${repoUrl}/compare`;
+	let commitsUrl = `/${repoUrl}/commits`;
+	let tree = '';
+	const urlParts = location.pathname.split('/');
+	if (urlParts[3] === 'tree') { // On repo page
+		tree = urlParts[4];
+	} else if (urlParts[3] === 'compare' && urlParts[4]) { // On compare page
+		if (urlParts[4].includes('...')) {
+			tree = urlParts[4].split('...')[0];
+		} else {
+			tree = urlParts[4];
+		}
+	} else if (urlParts[3] === 'commits') { // On commits page
+		tree = urlParts[4];
+	} else if (urlParts[3] === 'releases' && urlParts[4] === 'tag') { // On tag page
+		tree = urlParts[5];
+	}
```

Comment:
Maybe:

```js
if (isCompare() && location.pathname.includes('...')) {
	[ref] = pathnameParts[4].split('...');
} else {
	ref = pathnameParts[isTagsOrReleases() ? 5 : 4]
}
```

Reply:
Sorry, I think this broke it, not all pages have a ref. See if this works

```js
if (isCompare()) {
	[ref] = pathnameParts[4].split('...');
} else if (isTagsOrReleases()) {
	ref = pathnameParts[5]
} else if (['commits', 'blob', 'tree'].includes(pathnameParts[3])) {
	ref = pathnameParts[4]
}
```

Reply:
Fixed compare detection.
Added blame detection.
Prioritized order.

And moved ref detection to it's own function in utils.js. And also included tests.